Topics/Software_Tools/Git/Automated_Deployment.md

```diff
@@ -0,0 +1,66 @@
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The benefits of continuous deployment are faster development, less risky releases (since there are fewer changes per release), and constant improvements for customers. Issues with releases can be fixed more quickly, since development does not need to pause for releases. Releases are also not scheduled, so there is less pressure to finish everything by a certain deadline. However, just like delivery, feature flags **must** be a part of the coding culture, and application documentation must stay up to date with deployments. This is in order to ensure clarity on the included features in each release.
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;However, most importantly, both continuous delivery and continuous deployment can only occur if the new changes pass all the tests from continuous integration. This means that in order to successfully implement continuous delivery or deployment, you need to have an extensive set of test for your code that can verify that the changes are suitable for these environments. Otherwise, continuous delivery or deployment can lead to issues within staging and testing environments, or, worse, issues for customers.
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To summarize, automated deployment is used for continous deliver or deployment. In order for this to work, continuous integration needs to be used in order to extensively test the code and ensure it is acceptable to deploy to whichever environment (depending on if delivery or deployment is used). Although it requires a strong CI, it leads to faster development, less risky releases, and more customer feedback. 
+ <br>
+ <br>
+To learn more about CI/CD, please see the following article from Atlassian, an established software company from Australia: https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment
+ - To learn more about the difference between continous delivery and continuous deployment, see the following article: https://docs.github.com/en/actions/deployment/about-deployments/deploying-with-github-actions
+
+______
+
+ ### Examples of Common Automated Deployment Software
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Atlassian recommends using Bitbucket Pipelines (https://bitbucket.org/product/features/pipelines), which is their "CI/CD tool that's integrated into Bitbucket Cloud," their version of GitHub (https://bitbucket.org/product/). BitBucket provides extended CI/CD features that allow you to integrate with Jira, a task management software by Atlassian. However, you can implement CI/CD fairly easily with GitHub Actions.
```

Comment:
Bitbucket Pipelines is just one example of automated deployment software. Could you add a couple more? I don't think you need to explain them all in detail necessarily - just list the popular ones and some pros/cons of each.

Reply:
Added three more and included their features. I also added two resources that list more